Tasks/AzureCloudPowerShellDeployment/Utility.ps1

```diff
@@ -109,7 +121,18 @@ function Get-DiagnosticsExtensions($storageAccount, $extensionsPath, $storageAcc
                         {
                             try
                             {
-                                $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName
+                                $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName -ErrorAction Ignore
+                                if ($publicConfigStorageKey -eq $null)
+                                {
+                                    #if the storage account is not the classic one, get the primary key in another way
+                                    $publicConfigStorageKey = (Get-AzureRmStorageAccountKey -ResourceGroupName "$storageAccountResourceGroup" -Name "$publicConfigStorageAccountName").Value[0]
```

Comment:
AzureCloudPowerShellDeployment task is based on classic subscription, so I cmdlets  corresponding to ARM endpoints will fail here with classic subscription.